unkeyed/unkey
# fix(web): resolve overlapping banners

## What does this PR do?



Fixes #1042 

You can find the Loom preview [here](https://www.loom.com/share/665d0dc0794547a9a9d329e1a4526d04?sid=c16238a5-d0db-4c28-a726-09a7231dae7e). It should now handle the overlapping of banners whenever a new banner is added, removed or if it's height changes when the window is resized. PD:  it's not a perfect solution as it reorders the banners bottom fixed position as many times as the number of banners currently rendered.

This PR also resolves other non-reported issues:
1. Banners intended to persist were not working. Slight bug that mixed the behaviour between Banners with LS persist or with just RS.
2. The usage of `useLocalStorage` for the banners were causing hydration issues even the directive of `<Banner/>` was "use client". Solved it with the `initializeWithValue: false` arg option.

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*



## Type of change




## How should this be tested?



- **Test A: Handle overlapping banners**
1. Make sure you qualify for a `UsageBanner` (trial ending in 7 days, plan exceeded, etc).
2. Make the RateLimit or Audit banner popup (e.g. for the rate limit, delete the `ratelimit_ui` LS key and head to the RateLimit section)
3. You should see both banners properly displayed banner.
4. Before dismissing them, try resizing your window and verify that there are no overlapping banners (specially for long text banners such as the Audit one).

- **Test B: Make sure Banners with LS persistance are working property.** 
1. Delete the `ratelimit_ui` LS key (skip this step if there isn't one).
6. Head to the rate limit section and opt-in in case you haven't.
7. You should see the banner.
8. Try to dismiss it and reload the page. Test passes if the banner is not shown again unless you delete the mentioned LS key.

## Checklist



### Required


### Appreciated
